// Decompiled with JetBrains decompiler
// Type: SmartAssembly.SmartExceptionsWithUI.UnhandledExceptionHandlerWithUI1
// Assembly: o4tdggb3, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: FA0844EB-EF82-404C-B76C-C092ABDB9392
// Assembly location: C:\Users\Administrateur\Downloads\Virusshare.00090-msil\Trojan-Ransom.Win32.Blocker.bzqq-b74e3dc76fae79b75c26f27ecb85aae1eae4082a42bfa080f7805f32dedc9d67.exe

using SmartAssembly.SmartExceptionsCore;
using System;
using System.Reflection;
using System.Security;
using System.Windows.Forms;

namespace SmartAssembly.SmartExceptionsWithUI
{
  public class UnhandledExceptionHandlerWithUI1 : UnhandledExceptionHandler1
  {
    protected override void OnSecurityException(SecurityExceptionEventArgs1 e)
    {
      using (SecurityExceptionForm securityExceptionForm = new SecurityExceptionForm(e))
      {
        int num = (int) securityExceptionForm.ShowDialog();
      }
    }

    protected override void OnReportException(ReportExceptionEventArgs1 e)
    {
      using (ExceptionReportingForm exceptionReportingForm = new ExceptionReportingForm((UnhandledExceptionHandler1) this, e))
      {
        int num = (int) exceptionReportingForm.ShowDialog();
      }
    }

    protected override void OnFatalException(FatalExceptionEventArgs1 e)
    {
      int num = (int) MessageBox.Show(e.FatalException.ToString(), string.Format("{0} Unexpected Error", (object) "server2"), MessageBoxButtons.OK, MessageBoxIcon.Hand);
    }

    public static bool AttachApp()
    {
      try
      {
        UnhandledExceptionHandler1.AttachExceptionHandler((UnhandledExceptionHandler1) new UnhandledExceptionHandlerWithUI1());
        return true;
      }
      catch (SecurityException ex1)
      {
        try
        {
          try
          {
            typeof (Application).InvokeMember("EnableVisualStyles", BindingFlags.Static | BindingFlags.Public | BindingFlags.InvokeMethod, (Binder) null, (object) null, (object[]) null);
          }
          catch (MissingMethodException ex2)
          {
          }
          SecurityExceptionForm securityExceptionForm = new SecurityExceptionForm(new SecurityExceptionEventArgs1(string.Format("{0} cannot initialize itself because some permissions are not granted.\n\nYou probably try to launch {0} in a partial-trust situation. It's usually the case when the application is hosted on a network share.\n\nYou need to run {0} in full-trust, or at least grant it the UnmanagedCode security permission.\n\nTo grant this application the required permission, contact your system administrator, or use the Microsoft .NET Framework Configuration tool.", (object) "server2"), false));
          securityExceptionForm.ShowInTaskbar = true;
          int num = (int) securityExceptionForm.ShowDialog();
        }
        catch (Exception ex3)
        {
          int num = (int) MessageBox.Show(ex3.ToString(), string.Format("{0} Unexpected Error", (object) "server2"), MessageBoxButtons.OK, MessageBoxIcon.Hand);
        }
        return false;
      }
    }
  }
}
